packages/rust/build.sh

```diff
@@ -22,11 +21,12 @@ termux_step_pre_configure() {
 	termux_setup_cmake
 	termux_setup_rust
 
-	TERMUX_PKG_RECOMMENDS="rust-std-${CARGO_TARGET_NAME/_/-}"
+	# default rust-std package to be installed
+	TERMUX_PKG_DEPENDS+=", rust-std-${CARGO_TARGET_NAME/_/-}"
```

Comment:
Circular dependencies is allowed in apt, but I don't know whether it is allowed in pacman too. So ping @Maxython 

From https://www.debian.org/doc/debian-policy/ch-relationships.html :
> If there is a circular dependency among packages being installed or removed, installation or removal order honoring the dependency order is impossible, requiring the dependency loop be broken at some point and the dependency requirements violated for at least one package. Packages involved in circular dependencies may not be able to rely on their dependencies being configured before they themselves are configured, depending on which side of the break of the circular dependency loop they happen to be on. If one of the packages in the loop has no `postinst` script, then the cycle will be broken at that package; this ensures that all `postinst` scripts are run with their dependencies properly configured if this is possible. Otherwise the breaking point is arbitrary. Packages should therefore avoid circular dependencies where possible, particularly if they have `postinst` scripts.

Reply:
Do rust-std-aarch64-linux-android and siblings really need to depend on rust? 

It seems like it would be nice to avoid circular dependencies, our [buildorder.py script also checks for circular dependencies](https://github.com/termux/termux-packages/blob/master/scripts/buildorder.py#L307-L308) and errors if found (`./build-all.sh` currently does not work because of it)

Reply:
I suppose I can turn off the subpackage depend on parent. Will try to check build-all.sh.

Reply:
> Circular dependencies is allowed in apt, but I don't know whether it is allowed in pacman too.

Pacman supports circular dependencies, but it will warn about them during package installation/update.

Reply:
> It seems like it would be nice to avoid circular dependencies, our [buildorder.py script also checks for circular dependencies](https://github.com/termux/termux-packages/blob/master/scripts/buildorder.py#L307-L308) and errors if found (./build-all.sh currently does not work because of it)

Is running `./scripts/buildorder.py` and not seeing rust there means no issue?
`./scripts/buildorder.py` only checks on single line `TERMUX_PKG_DEPENDS` which in this case (split `TERMUX_PKG_DEPENDS` onto different lines) can fail to detect and passed. Not sure that's a feature or bug.
`./scripts/buildorder.py` also dont check `TERMUX_SUBPKG_DEPEND_ON_PARENT`...
`./scripts/buildorder.py` also erroneously prints `*-static` subpackages when the whole package has issue (which I think `buildorder.py` does try to ignore `*-static`)?
I believe `build-all.sh` issue should have tracker issue on its own...

> Do rust-std-aarch64-linux-android and siblings really need to depend on rust?

I think subpackages should be given leeway for cyclic dependencies on parents and vice versa. This also helps to stop having mixing different rust version packages and subpackages from happening.
If I were back to use `TERMUX_PKG_RECOMMENDS` then I guess `pacman` users have to manually install rust-std-* ...
And since no `postinst` is used, it should be safe for this case as pointed in Debian packaging guidelines...

Apparently `libfcft` is causing problem for not able to start `./scripts/buildorder.py packages/rust` probably need to be done in a separate PR